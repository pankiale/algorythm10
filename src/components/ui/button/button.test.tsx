import renderer from 'react-test-renderer';
import {fireEvent, render, screen} from "@testing-library/react";
import {Button} from "./button";

const buttonTextStub = 'Текст кнопки'

describe("button tests", () => {
    it('button text', () => {
        const tree = renderer
            .create(<Button text={buttonTextStub}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('button no text', () => {
        const tree = renderer
            .create(<Button/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('button disabled', () => {
        const tree = renderer
            .create(<Button disabled/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('button loading', () => {
        const tree = renderer
            .create(<Button isLoader/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('button onclick', () => {
        const mockOnClick = jest.fn();
        render(<Button text={"testText"} onClick={mockOnClick}/>)
        const button = screen.getByText('testText');
        fireEvent.click(button);
        expect(mockOnClick).toHaveBeenCalled();
    })
})