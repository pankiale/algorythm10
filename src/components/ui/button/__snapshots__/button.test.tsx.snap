// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button tests button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button tests button loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button tests button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст кнопки
  </p>
</button>
`;

exports[`button tests button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
